lib/utils/isStandardSyntaxDeclaration.js

```diff
@@ -19,18 +12,6 @@ module.exports = function isStandardSyntaxDeclaration(decl) {
 	const prop = decl.prop;
```

Comment:
`isStandardSyntaxDeclaration` is used in at least 7 rules. Are we sure this is not gonna cause regressions?  
Is being more lax really covered by tests already?